p2p/service.go

```diff
@@ -133,36 +133,39 @@ func (s *Service) receiveCoreMessages() {
 		// receive message from core service
 		msg := <-s.msgRec
 
-		// check if non-status message
-		if msg.GetType() != StatusMsgType {
-
-			log.Trace(
-				"Broadcasting message from core service",
-				"host", s.host.id(),
-				"type", msg.GetType(),
-			)
-
-			// broadcast message to connected peers
-			s.host.broadcast(msg)
-
-		} else {
-
-			// check if status is enabled
-			if !s.noStatus {
+		log.Trace(
+			"Broadcasting message from core service",
+			"host", s.host.id(),
+			"type", msg.GetType(),
+		)
 
-				// update host status message
-				s.status.setHostMessage(msg)
-			}
-		}
+		// broadcast message to connected peers
+		s.host.broadcast(msg)
 	}
 }
 
 // handleConn starts processes that manage the connection
 func (s *Service) handleConn(conn network.Conn) {
-
 	// check if status is enabled
 	if !s.noStatus {
 
+		// get latest block header from block state
+		latestBlock := s.cfg.BlockState.LatestHeader()
+
+		// update host status message
+		msg := &StatusMessage{
+			ProtocolVersion:     s.cfg.ProtocolVersion,
+			MinSupportedVersion: s.cfg.MinSupportedVersion,
+			Roles:               s.cfg.Roles,
+			BestBlockNumber:     latestBlock.Number.Uint64(),
+			BestBlockHash:       latestBlock.Hash(),
+			GenesisHash:         latestBlock.StateRoot,
+			ChainStatus:         []byte{0}, // TODO
+		}
+
+		// update host status message
+		s.status.setHostMessage(msg)
+
```

Comment:
Does it make sense to load a fresh status message here on connect?

Reply:
I think only reloading it upon connecting makes sense, however, there's a pretty high chance we might connect to multiple peers without our state ever changing. maybe there should be a way to check if the status message should change from previously (eg. if there's a new block), then if so, update it

Reply:
we could open another issue for that, since this PR is already  somewhat large

Reply:
Yea, I guess it might not need to be updated so frequently. I will open an issue to investigate and address. Might also be nice to add a test or make note when we eventual write and run network simulations.

Reply:
#602